### PR TITLE
rand: make pseudo-random variants a bit more variable

### PR DESCRIPTION
**Pull request by mrc0mmand**
Until now all pseudo-random variants were always strings. Since a true
variant can contain any _complete_ type, let's use at least any
_basic_ type, for now. Extending this to any complete type must wait
until we can generate arbitrary signatures.

### CODE REVIEW

**Comment by evverx** [comment]
Looks like it should be rebased.

I can't seem to figure out what's going on here :-) so let me try it first. Does it have something to do with https://github.com/dbus-fuzzer/dfuzzer/issues/52?

**Comment by mrc0mmand** [comment]
> Looks like it should be rebased.
> 
> I can't seem to figure out what's going on here :-) so let me try it first. Does it have something to do with #52?

It's definitely a part of preparations to fix #52, but more importantly it's just extending coverage of what 'variant' truly is - it can be any _single complete_ D-Bus type, the only difference is that the signature is part of the value. So `v` in signature might mean `s` as well as `(isg)`, etc.

https://dbus.freedesktop.org/doc/dbus-specification.html#container-types

But until now, we basically assumed that every variant is a string. This PR tries to extend this a bit to cover all _basic_ types. Once we're able to generate arbitrary signatures, we can easily extend this to generate any complete type for a variant.

**Comment by mrc0mmand** [comment]
> @mrc0mmand I tested this PR and it seems to be working but before I merge it I wonder how dfuzzer gets updated on CentOS CI? If it picks up all new commits I think new PRs should be merged more carefully so as not to break the CI.

The Arch Linux images get updated twice a week (Monday & Thursday), and due to previous experience they're "gated", meaning that the CI will generate a new updated image (including newer dfuzzer) and runs all related CI jobs on it. If any of the jobs fail, I'll get an email and the image won't get pushed into "production".

tl;dr even if we screw something up here, it won't affect the systemd CI jobs, but I'll definitely notice it once CI attempts to update the image :-)

**Review by evverx — APPROVED** [review]
LGTM. Thanks!

I'll go ahead and merge it to let Coverity review it as well
